### PR TITLE
feat: bump for react 17

### LINKED ISSUE #109: React 17 support
`react-helmet-async` currently supports react 16:

https://github.com/staylor/react-helmet-async/blob/5c708c1554cfac6e7384484eb9c544e9709cd54c/package.json#L57-L60

Please also add support for [React 17](https://reactjs.org/blog/2020/10/20/react-v17.html).

### PR DESCRIPTION
**Pull request by levigunz**
May want to wait for the Enzyme Adapter for React 17 before merging this. I don't believe there will be any changes that effect this project though.

closes #109

### CODE REVIEW

**Comment by bcomnes** [comment]
I would like to see the peer dep land sooner than later so I can clean up these warnings.  React helmet still working great with react 17.

**Comment by levigunz** [comment]
We have been using in production since 17rc as well with no issues. I just hate those pesky warnings!

@staylor any chance we can get a merge and tag release 🚢

**Comment by bcomnes** [comment]
Looks like master has changed and this PR needs to be rebased.

**Comment by bcomnes** [comment]
Happy new year! Is there anything we can do to help land this?

**Comment by levigunz** [comment]
Sorry about that, auto-closed this when rebasing. Opening back up with resolved changes.

File: package.json [hunk 1]
```diff
@@ -52,13 +52,13 @@
     "microbundle": "0.12.4",
     "prettier": "2.1.2",
     "raf": "3.4.1",
-    "react": "16.14.0",
-    "react-dom": "16.14.0",
+    "react": "17.0.1",
```
Comment (Hypnosphi) [inline package.json:55]: devDependencies should probably stay on 16 until enzyme supports 17, otherwise the tests won't pass

Comment (levigunz) [reply package.json:55]: Tests will pass with 17 as well, but agreed that waiting for 17 adapter would be nice.

Comment (Hypnosphi) [reply package.json:55]: So let's maybe update only the peer dependencies for now?

Comment (theskillwithin) [reply package.json:55]: Yes please

**Comment by theskillwithin** [comment]
can't use npm 7 because this isnt merged yet

**Comment by basvandriel** [comment]
@staylor Any chance this can get merged? Only for the peer dependencies please!

File: package.json (hunk 1 above)
Comment (wojtekmaj) [reply package.json:55]: peerDependencies updated in #119

File: package.json [hunk 2]
```diff
@@ -52,13 +52,13 @@
...
+    "react-dom": "17.0.1",
     "rimraf": "3.0.2"
   },
   "peerDependencies": {
-    "react": "^16.6.0",
-    "react-dom": "^16.6.0"
+    "react": "^16.6.0 || ^17.0.0",
+    "react-dom": "^16.6.0 || ^17.0.0"
```
Comment (staylor) [inline package.json:61]: these have been updated in: https://www.npmjs.com/package/react-helmet-async/v/1.0.8

**Pull request closed without merging**